src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,259 @@
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        elif "host_storage_map" in result.keys():
+            for value in list(result['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        else:
+            self.log.info("##Unsupported result dictionary %s", result)
+        return False
+
+    def reset_fault_device(self, device):
+        """Call dmg storage led identify to reset the device.
+
+        Args:
+            device (str): device to reset
+
+        Returns:
+            list: a list of any errors detected when removing the pool
+        """
+        error_list = []
+        try:
+            get_dmg_response(self, self.dmg.storage_led_identify, reset=True, ids=device)
+        except TestFail as error:
+            self.log.info("#  %s", error)
+            error_list.append("Error resetting device {}: {}".format(device, error))
+        return error_list
+
+    def run_ior_with_nvme_fault(self, oclass=None):
+        """Perform ior write with nvme fault testing.
+
+        Test steps:
+            1. Start the DAOS servers.
+            2. Verify the drive status LED.
+            3. Create the pool and container with RF.
+            4. While IOR (EC+GX) in progress put single drive to fault mode.
+            5. IOR will halt until rebuild is finished.
+            6. IOR should complete without any error.
+            7. Drive status LED should be solid Amber ON indicating it’s faulty.
+            8. Verify IOR data and check container.
+            9. Replace the same drive back.
+            10. Drive status LED should be off indicating good device is plugged-in.
+        Args:
+            oclass (str, optional): object class (eg: RP_2G8, S1,etc). Defaults to None
+        """
+        # 1.
+        self.log_step("Bring up server and reset led of all devices.")
+        if oclass is None:
+            oclass = self.ior_cmd.dfs_oclass.value
+        device_info = get_storage_query_device_info(self, self.dmg)
+        devices = [device['uuid'] for device in device_info]
+        self.log.info("Device information")
+        self.log.info("==================")
+        for index, entry in enumerate(device_info):
+            self.log.info("Device %s:", index)
+            for key in sorted(entry):
+                self.log.info("  %s: %s", key, entry[key])
+        test_dev = devices[0]
+        for device in devices:
+            self.run_vmd_led_identify(device, reset=True)
```

Comment:
We can do this instead which means we don't need to modify `self.run_vmd_led_identify()` and its easier to maintain:
```suggestion
            get_dmg_response(self, self.dmg.storage_led_identify, reset=True, ids=device)
```